src/Peachpie.Library/Reflection/ReflectionType.cs

```diff
@@ -11,19 +11,24 @@ namespace Pchp.Library.Reflection
     [PhpType(PhpTypeAttribute.InheritName), PhpExtension(ReflectionUtils.ExtensionName)]
     public class ReflectionType
     {
-        public virtual bool allowsNull() { throw new NotImplementedException(); }
+        public string Name { get; set; }
+        public bool AllowsNull { get; set; }
+        public bool BuiltIn { get; set; }
```

Comment:
The class should not have properties `Name`, `AllowsNull` and `BuiltIn` visible (https://www.php.net/manual/en/class.reflectiontype.php)

They should be marked `internal` or `internal protected`.